include/regs.h

```diff
@@ -49,13 +49,14 @@
 #define R_RUN_SPEED_LIMIT                        REG(45)
 #define R_ENABLE_ARENA_DBG                       SREG(0)
 #define R_UPDATE_RATE                            SREG(30)
+#define R_GRAPH_TASKSET00_FLAGS                  SREG(33)
 #define R_ENABLE_AUDIO_DBG                       SREG(36)
 #define R_FB_FILTER_TYPE                         SREG(80)
 #define R_FB_FILTER_PRIM_COLOR(c)                SREG(81 + (c))
 #define R_FB_FILTER_A                            SREG(84)
 #define R_FB_FILTER_ENV_COLOR(c)                 SREG(85 + (c))
 #define R_ENABLE_FB_FILTER                       SREG(88)
-#define R_PAUSE_MENU_MODE                        SREG(94)
+#define R_PAUSE_BG_PRERENDER_STATE               SREG(94) // `PauseBgPreRenderState`
```

Comment:
I don't really understand the reasoning behind this rename. `R_PAUSE_MENU_MODE` handles more than just the prerender state, since it actually governs how and when the pause menu gets loaded and drawn on the screen (partially at least), and it would exist in some form even if the prerender wasn't a thing.
Some of its values should probably be named around the prerender steps, or documented with them, but I don't think it makes sense to name the reg entirely around the prerender part.

Reply:
Everywhere it is used, `R_PAUSE_BG_PRERENDER_STATE` is about the state of the pause menu's background prerender. Mainly, in z_play all the states refer to the step making the prerender is at. In other spots it's about whether the pause menu may draw, which relies entirely on the prerender being done

All values of `R_PAUSE_BG_PRERENDER_STATE` are tied to the state of the pause background prerender so changing the reg's name seemed natural

Note the actual pause state `PauseContext.state` is something else (and a way bigger enum)

Reply:
Yeah in my mind `pauseCtx.state` has always conflicted with `R_PAUSE_MENU_MODE` in name. I am currently feeling like it would be beneficial to be specific about what its for.

In my opinion the only usage of the mode that can be argued is NOT for the bg processing is `R_PAUSE_MENU_MODE >= 3` which says that the pause menu is active and can draw. But as dragorn has said above, you can also look at it from the viewpoint of "bg processing is done, so pause menu can draw now".

Reply:
` since it actually governs how and when the pause menu gets loaded and drawn on the screen (partially at least), and it would exist in some form even if the prerender wasn't a thing`

as far as I can tell, overlay loading is determined only by `pauseCtx.state`. It does control when it can draw, but as mentioned above its really only delaying drawing until the pause bg is done. 
Its purpose for existing is to control the flow of drawing in Play_Draw in relation to the background processing. So that it can for example exclude the hud in its screenshot. If the bg didnt exist i believe everything could be simplified to be controlled by the state variable

Reply:
as a closing note on my spiel, I will note that we do know its original name does agree with calling it a generic mode: `osSyncPrintf("PR_KAREIDOSCOPE_MODE=%d\n", R_PAUSE_MENU_MODE);`
But yea even still I think being more specific is still beneficial here

Reply:
At the very least this isn't a "mode" that can change at will to have things behave differently, it describes what step the prerender is at (so "state"/...)

It's not really about UI (besides allowing to draw some as part of drawing the pause screen at all) so I wouldn't name it with the current "menu"

that leaves `R_PAUSE_STATE`, which is too generic, and knowing this deals (at 100% afaict) with drawing the prerender background well -> `R_PAUSE_BG_PRERENDER_STATE`

Reply:
Worth noting as well that MM adds a similar variable for its pictobox prerender that aligns with the pause menu one really closely. They both are pretty clearly about the state of the prerender as its being taken and processed

Reply:
Right, I've thought about using "kaleido" as a term encompassing both the pause menu and the game over thing, but it would end up just using kaleido for everything with not much actually being specific to pause or gameover

It's just really weird that these two things are not separated... And since gameover is comparatively very simple I'd rather just call everything "pause" yeah but this is debatable